packages/rainbowkit/src/wallets/connectorsForWallets.ts

```diff
@@ -1,151 +1,109 @@
-          connectionMethods.qrCode &&
-          !isMobile()
-        ) {
-          const { chains, options } = connector;
-
-          walletConnectModalConnector = new WalletConnectConnector({
-            chains,
-            options: {
-              ...options,
-              showQrModal: true,
-            },
-          });
-
-          connectors.push(walletConnectModalConnector);
+      if (typeof wallet.hidden === 'function') {
+        potentiallyHiddenWallets.push(walletListItem);
+      } else {
+        visibleWallets.push(walletListItem);
+      }
+    });
+  });
+
+  // We process the known visible wallets first so that the potentially
+  // hidden wallets have access to the complete list of resolved wallets
+  const walletListItems: WalletListItem[] = [
+    ...visibleWallets,
+    ...potentiallyHiddenWallets,
+  ];
+
+  walletListItems.forEach(
+    ({ createConnector, groupIndex, groupName, hidden, ...walletMeta }) => {
+      if (typeof hidden === 'function') {
+        // Run the function to check if the wallet needs to be hidden
+        const isHidden = hidden();
+
+        // If wallet is hidden, bail out and proceed to the next list item
+        if (isHidden) {
+          return;
         }
-
-        const walletInstance: WalletInstance = {
-          connector,
-          groupIndex,
-          groupName,
-          index,
-          walletConnectModalConnector,
-          ...walletMeta,
-          ...connectionMethods,
+      }
+
+      const params = (
+        // For now we should only use these as the additional parameters
+        additionalParams?: Pick<
+          WalletOptionsParams['rkDetails'],
+          'showQrModal' | 'isWalletConnectModalConnector'
+        >,
+      ) => {
+        return {
+          rkDetails: omitUndefinedValues({
+            ...walletMeta,
+            groupIndex,
+            groupName,
+            isRainbowKitConnector: true,
+            ...additionalParams,
+          }),
         };
+      };
 
-        // We maintain an array of all wallet instances
-        // so they can be passed to the `hidden` function
-        // used by generic fallback wallets
-        walletInstances.push(walletInstance);
-
-        if (!connectors.includes(connector)) {
-          connectors.push(connector);
-
-          // Reset private wallet list the first time we see
-          // a connector to avoid duplicates after HMR,
-          // otherwise we'll keep pushing wallets into
-          // the old list. This is happening because we're
-          // re-using the WalletConnectConnector instance
-          // so the wallet list already exists after HMR.
-          connector._wallets = [];
-        }
+      const isWalletConnectConnector = walletMeta.id === 'walletConnect';
+
+      if (isWalletConnectConnector) {
+        connectors.push(
+          createConnector(
+            params({ showQrModal: true, isWalletConnectModalConnector: true }),
+          ),
+        );
+      }
+
```

Comment:
The wallet connect modal also needed to be included. Made a cool trick here to push connector again just with `showQrModal` and then in `useWalletConnectors` hook we can use that connector to open the wallet connect modal.